tracer/test/benchmarks/Benchmarks.Trace/Asm/AppSecWafBenchmark.cs

```diff
@@ -78,10 +78,31 @@ public IEnumerable<NestedMap> Source()
         yield return MakeNestedMap(1000);
     }
 
-    private static NestedMap MakeNestedMap(int nestingDepth)
+    public IEnumerable<NestedMap> SourceWithAttack()
+    {
+        yield return MakeNestedMap(10, true);
+        yield return MakeNestedMap(100, true);
+        yield return MakeNestedMap(1000, true);
+    }
+
+    private static NestedMap MakeNestedMap(int nestingDepth, bool withAttack = false)
```

Comment:
You should update the `NestedMap.ToString()` override. BenchmarkDotNet (and the benchmark comparison) uses it to find "unique" runs, so currently you will get a name clash. Also be aware that if you _change_ the name of an existing run you'll break previous comparisons, so I suggest something like this:

```csharp
public record NestedMap(Dictionary<string, object> Map, int NestingDepth, bool IsAttack)
{
    public override string ToString() => IsAttack ? $"NestedMap ({NestingDepth}, Attack) : $"NestedMap ({NestingDepth})";
}
```

Reply:
thanks very much, updated in [d4e52b3](https://github.com/DataDog/dd-trace-dotnet/pull/4578/commits/d4e52b3e998fd365ac51759c52c47b8369cefd00)